From: Alberts Muktupavels <alberts.muktupavels@gmail.com>
Date: 2022-09-20 15:40:22 +0300
Subject: Avoid segfault in invalidate_icons
References: boo#1204157 lp#1990263 glgo#GNOME/libwnck!46
Upstream: submitted

Screens variable won't be initialized if default icon sizes are
set/changed before using wnck_screen_get_default

---
 libwnck/wnck-handle.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/libwnck/wnck-handle.c	2022-11-09 14:54:06.445579490 +0100
+++ b/libwnck/wnck-handle.c	2022-11-09 14:54:45.522267274 +0100
@@ -70,6 +70,9 @@
   Display *xdisplay;
   int i;
 
+  if (self->screens == NULL)
+    return;
+
   xdisplay = _wnck_get_default_display ();
 
   for (i = 0; i < ScreenCount (xdisplay); ++i)
